worktree-helper.sh: localdev proxy YAML write fails on macOS without PyYAML + BSD sed regex error (non-fatal)

## What

When running `worktree-helper.sh add feature/<branch>` on macOS, two non-fatal errors print to stderr immediately after port allocation:

```
[ERROR] YAML corruption in /Users/<user>/.local-dev-proxy/conf.d/<repo>--<branch>.yml (Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'yaml') — removing
Localdev branch route creation failed (non-fatal)
sed: 1: "s|.*[:/]([^/]+/[^/]+?)( ...": RE error: repetition-operator operand invalid
```

The worktree itself is created correctly; only the localdev-proxy route registration fails. Output is misleadingly labelled `[ERROR]` for a non-fatal path, and BSD sed (macOS default) chokes on the slug-extraction regex regardless of PyYAML state.

## Why

1. `python3 -c 'import yaml'` is invoked inline but PyYAML isn't guaranteed on macOS — many users only have stdlib python3. Should fall back to a pure-shell/stdlib writer, or skip with a single `[INFO]` log when PyYAML missing (not an `[ERROR]` traceback).
2. The `sed` slug-extraction expression uses Perl-flavoured regex syntax (`+?` non-greedy) that BSD sed (macOS default) doesn't support. Should use `sed -E` with portable POSIX-ERE, or bash parameter expansion `${var##*/}`.

## How

### Files Scope

- `EDIT: .agents/scripts/worktree-helper.sh` — localdev-proxy YAML write block: detect PyYAML availability before invocation, skip with `[INFO]` log when missing.
- `EDIT: .agents/scripts/worktree-helper.sh` — slug-extraction `sed`: switch to portable POSIX-ERE (`sed -E`) or bash parameter expansion. Current regex `s|.*[:/]([^/]+/[^/]+?)(...)` uses non-greedy `+?` which is a GNU/Perl extension.

### Verification

```
worktree-helper.sh add feature/test-localdev
# On macOS without PyYAML: should print no [ERROR] lines.
# Either proxy YAML is written cleanly, or skipped with one [INFO] log.
# No "sed: RE error: repetition-operator operand invalid" output.
```

### Acceptance

- [ ] No `ModuleNotFoundError` traceback on macOS without PyYAML
- [ ] No `sed: RE error` on macOS BSD sed
- [ ] Worktree creation continues to succeed (already does — this is purely cosmetic + UX)
- [ ] If localdev-proxy YAML write is skipped, log level is `INFO` not `ERROR`

## Reproduction

Reproduced on macOS Darwin 2026-04-29 during ILDS interactive session creating `feature/t153-remove-header-pricing-line` worktree. macOS default python3 (no PyYAML), default `/usr/bin/sed` (BSD).

## Tier

`tier:simple` — one file, two narrow edits (PyYAML guard + sed portability).
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.13.10 plugin for [OpenCode](https://opencode.ai) v1.14.29 with claude-opus-4-7 spent 3h 59m and 16,728 tokens on this with the user in an interactive session.

--- a/.agents/scripts/localdev-helper-branch.sh
+++ b/.agents/scripts/localdev-helper-branch.sh
@@ -175,11 +175,16 @@ tls:
       keyFile: /certs/${app_domain}+1-key.pem
 YAML
 
-	# Validate: reject files containing ANSI escape codes or non-parseable YAML
+	# Validate: reject files containing ANSI escape codes or non-parseable YAML.
+	# PyYAML is required for YAML parse validation; if absent, skip with INFO
+	# (MacOS stdlib python3 does not include PyYAML — GH#21782).
 	if command -v python3 >/dev/null 2>&1; then
-		local py_err
-		py_err="$(
-			python3 - "$route_file" 2>&1 <<'PYEOF'
+		if ! python3 -c "import yaml" 2>/dev/null; then
+			print_info "PyYAML not available — skipping YAML syntax validation for $route_file"
+		else
+			local py_err
+			py_err="$(
+				python3 - "$route_file" 2>&1 <<'PYEOF'
 import sys, yaml
 path = sys.argv[1]
 with open(path, 'rb') as fh:
@@ -193,12 +198,13 @@ except yaml.YAMLError as e:
     print(f"YAML parse error: {e}")
     sys.exit(2)
 PYEOF
-		)"
-		local py_exit=$?
-		if [[ "$py_exit" -ne 0 ]]; then
-			print_error "YAML corruption in $route_file ($py_err) — removing"
-			rm -f "$route_file"
-			return 1
+			)"
+			local py_exit=$?
+			if [[ "$py_exit" -ne 0 ]]; then
+				print_error "YAML corruption in $route_file ($py_err) — removing"
+				rm -f "$route_file"
+				return 1
+			fi
 		fi
 	fi
 	print_success "Created branch route: $route_file"

--- a/.agents/scripts/localdev-helper-routes.sh
+++ b/.agents/scripts/localdev-helper-routes.sh
@@ -74,11 +74,16 @@ tls:
       keyFile: /certs/${domain}+1-key.pem
 YAML
 
-	# Validate: reject files containing ANSI escape codes or non-parseable YAML
+	# Validate: reject files containing ANSI escape codes or non-parseable YAML.
+	# PyYAML is required for YAML parse validation; if absent, skip with INFO
+	# (MacOS stdlib python3 does not include PyYAML — GH#21782).
 	if command -v python3 >/dev/null 2>&1; then
-		local py_err
-		py_err="$(
-			python3 - "$route_file" 2>&1 <<'PYEOF'
+		if ! python3 -c "import yaml" 2>/dev/null; then
+			print_info "PyYAML not available — skipping YAML syntax validation for $route_file"
+		else
+			local py_err
+			py_err="$(
+				python3 - "$route_file" 2>&1 <<'PYEOF'
 import sys, yaml
 path = sys.argv[1]
 with open(path, 'rb') as fh:
@@ -92,12 +97,13 @@ except yaml.YAMLError as e:
     print(f"YAML parse error: {e}")
     sys.exit(2)
 PYEOF
-		)"
-		local py_exit=$?
-		if [[ "$py_exit" -ne 0 ]]; then
-			print_error "YAML corruption in $route_file ($py_err) — removing"
-			rm -f "$route_file"
-			return 1
+			)"
+			local py_exit=$?
+			if [[ "$py_exit" -ne 0 ]]; then
+				print_error "YAML corruption in $route_file ($py_err) — removing"
+				rm -f "$route_file"
+				return 1
+			fi
 		fi
 	fi
 	print_success "Created Traefik route: $route_file"

--- a/.agents/scripts/loop-common.sh
+++ b/.agents/scripts/loop-common.sh
@@ -1086,8 +1086,11 @@ loop_handle_workflow_push_failure() {
 		local remote_url
 		remote_url=$(git remote get-url origin 2>/dev/null || echo "")
 		if [[ -n "$remote_url" ]]; then
-			# Extract owner/repo from SSH or HTTPS URL
-			repo_slug=$(echo "$remote_url" | sed -E 's#.*[:/]([^/]+/[^/]+?)(\.git)?$#\1#')
+			# Extract owner/repo from SSH or HTTPS URL.
+			# Strip .git first, then use POSIX ERE (avoids non-greedy +? which
+			# BSD sed on macOS does not support — GH#21782).
+			local _remote_no_git="${remote_url%.git}"
+			repo_slug=$(echo "$_remote_no_git" | sed -E 's#.*[:/]([^/]+/[^/]+)$#\1#' || true)
 		fi
 	fi
 

--- a/.agents/scripts/runtime-audit-rules/counter-trend-delta.sh
+++ b/.agents/scripts/runtime-audit-rules/counter-trend-delta.sh
@@ -79,8 +79,12 @@ runtime_audit_check() {
 	first_counter=$(printf '%s\n' "$regressions" | head -1 | cut -f1)
 
 	local title="runtime-audit: counter regression detected (\`${first_counter}\`)"
-	local body
-	body=$(cat <<MARKDOWN
+	# Build body via temp file — avoids heredoc-inside-$() which breaks Bash 3.2
+	# (single quotes in heredoc content confuse the Bash 3.2 $() nesting parser).
+	# GH#21782: pre-existing Bash 3.2 compat fix bundled with PyYAML/sed PR.
+	local _body_tmp
+	_body_tmp=$(mktemp)
+	cat >"$_body_tmp" <<MARKDOWN
 ## Task
 
 A pulse counter has regressed sharply in the last $((window / 3600))h compared to the prior $((window / 3600))h window. This is the kind of trend shift that the supervisor LLM cycle does not detect because it never reads \`pulse-stats.json\` directly — it relies on issue and PR state.
@@ -118,7 +122,9 @@ The regression block for the cited counters should no longer appear.
 
 <!-- aidevops:generator=runtime-audit detector=counter-trend-delta -->
 MARKDOWN
-)
+	local body
+	body=$(cat "$_body_tmp")
+	rm -f "$_body_tmp"
 
 	jq -n --arg id "counter-trend-delta" --arg title "$title" --arg body "$body" \
 		'{id: $id, title: $title, body: $body}'

